net/instaweb/rewriter/css_combine_filter.cc

```diff
@@ -504,6 +504,11 @@ bool CssCombineFilter::CssCombiner::WritePiece(
       input_url, combination->resolved_base(), contents, writer, handler)) {
     case RewriteDriver::kNoResolutionNeeded:
       ret = writer->Write(contents, handler);
+      if (!contents.ends_with("\n") && ret) {
+        // Ensure that we add a new line at the end, to make sure that any
+        // unopened strings get closed in a predictable manner.
+        ret = writer->Write("\n", handler);
+      }
```

Comment:
I notice (from the tests) you seem to be adding the trailing newline unconditionally at the end of a CSS block.  What would it take to just insert it in the between pairs of CSS and / or avoid the trailing newline?